### ISSUE TITLE
Ensure consistency between Results and rounds data

### ISSUE THREAD

**Issue by SAuroux** [thewca/worldcubeassociation.org#5514]
**Problem description**
Since the beginning of 2019, the consistency between Results and rounds data is ensured by results validation, but only _at the time competition results are posted_.
Now, WRT is currently working on a project where we are backfilling rounds data based on what can be derived from Results data for all past competitions. Subsequently, we will have complete rounds data for all competitions that is perfectly consistent with the data available in the Results table.

The question that remains is: **What can be done to maintain this consistency?**

**Solution discussion and issue purpose**

One specific idea to help ensure this consistency is locking events management for competitions with posted results and has already been described in a separate issue #5513.

The purpose of this issue is to provide room for a general discussion and collecting more ideas about this consistency could be maintained further.

**Comment by danieljames-dj** [comment]
I agree with closing this issue and maybe when someone comes up with an additional ideas for results validators or sanity checks, they can create a new issue for that particular case (like we have for advancement condition #5587).
